src/FunctionLoader.ts

```diff
@@ -7,21 +7,20 @@ import { loadScriptFile } from './loadScriptFile';
 import { PackageJson } from './parsers/parsePackageJson';
 import { InternalException } from './utils/InternalException';
 import { nonNullProp } from './utils/nonNull';
+import { RegisteredFunction } from './WorkerChannel';
 
-export interface IFunctionLoader {
+export interface ILegacyFunctionLoader {
     load(functionId: string, metadata: rpc.IRpcFunctionMetadata, packageJson: PackageJson): Promise<void>;
-    getRpcMetadata(functionId: string): rpc.IRpcFunctionMetadata;
-    getCallback(functionId: string): FunctionCallback;
+    getFunction(functionId: string): RegisteredFunction;
 }
 
-interface LoadedFunction {
-    metadata: rpc.IRpcFunctionMetadata;
-    callback: FunctionCallback;
+interface LegacyRegisteredFunction extends RegisteredFunction {
     thisArg: unknown;
 }
 
-export class FunctionLoader implements IFunctionLoader {
-    #loadedFunctions: { [k: string]: LoadedFunction | undefined } = {};
+// todo rename this file after PR is reviewed
+export class LegacyFunctionLoader implements ILegacyFunctionLoader {
+    #loadedFunctions: { [k: string]: LegacyRegisteredFunction } = {};

```

Comment:
Did you enable the typescript rule you told me about to make sure `| undefined` is implied here?

Reply:
Added the `| undefined` back - not sure why I removed it in the first place. I have not enabled the rule